### PR TITLE
Spelling scripts

### LINKED ISSUE #30894: Fix typos in scripts
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

possibly all of them?

### What changes are you suggesting?

Fixing a bunch of typos...

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jsoref**
### Why:

Closes: #30894

### Check off the following:

- [ ] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [ ] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by jsoref** [comment]
Fwiw, https://github.com/github/docs/actions/runs/7388828032/job/20100431728 is going to ❌ -- see https://github.com/github/docs/issues/30896

File: src/content-linter/tests/unit/liquid-data-tags.js [hunk 1]
```diff
@@ -23,8 +23,8 @@ describe(liquidDataReferencesDefined.names.join(' - '), () => {
       '{% data variables.no-file %}',
       // Variables even when they exist can't be nested
       '{% data variables.location.foo.bar %}',
-      '{% data resuables.gated-features.empty %}',
-      '{% data resuables.no-file %}',
+      '{% data reusables.gated-features.empty %}',
+      '{% data reusables.no-file %}',
```
Comment (jsoref) [inline src/content-linter/tests/unit/liquid-data-tags.js:27]: ???

File: src/frame/middleware/api.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ router.use('/pageinfo', pageInfo)
 // The purpose of this is for convenience to everyone who runs this code
 // base locally but don't have an Elasticsearch server locally.
 // In production, this env var is always set but perhaps in a writer's
-// local laptop, they don't have an Elasitsearch. Neither a running local
+// local laptop, they don't have an Elasticsearch. Neither a running local
```
Comment (jsoref) [inline src/frame/middleware/api.js:20]: brand

**Comment by nguyenalex836** [comment]
@jsoref Thanks so much for opening a PR! I'll get this triaged for review ✨

File: src/frame/middleware/context/generic-toc.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import findPageInSiteTree from '#src/frame/lib/find-page-in-site-tree.js'
 
 // This module adds either flatTocItems or nestedTocItems to the context object for
-// product, categorie, and map topic TOCs that don't have other layouts specified.
-// They are rendered by includes/generic-toc-flat.html or inclueds/generic-toc-nested.html.
+// product, category, and map topic TOCs that don't have other layouts specified.
```
Comment (jsoref) [inline src/frame/middleware/context/generic-toc.js:4]: It's possible it meant categories, but...

File: src/frame/middleware/reload-tree.js [hunk 4]
```diff
@@ -36,7 +36,7 @@ export default async function reloadTree(req, res, next) {
 
   const warmed = await warmServer()
   // For all the real English content, this usually takes about 30-60ms on
-  // an Intel MacbookPro.
+  // an Intel MacBook Pro.
```
Comment (jsoref) [inline src/frame/middleware/reload-tree.js:39]: brand.

File: src/redirects/lib/precompile.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ import { latest } from '#src/versions/lib/enterprise-server-releases.js'
 const EXCEPTIONS_FILE = './src/redirects/lib/static/redirect-exceptions.txt'
 
 // This function runs at server warmup and precompiles possible redirect routes.
-// It outputs them in key-value pairs within a neat Javascript object: { oldPath: newPath }
+// It outputs them in key-value pairs within a neat JavaScript object: { oldPath: newPath }
```
Comment (jsoref) [inline src/redirects/lib/precompile.js:8]: brand

File: src/search/middleware/es-search.js [hunk 6]
```diff
@@ -185,8 +185,8 @@ function getMatchQueries(query, { usePrefixSearch, fuzzy }) {
   const BOOST_EXPLICIT = 3.5
   // Number doesn't matter so much but just make sure it's
   // boosted low. Because we only really want this to come into
-  // play if nothing else matches. E.g. a search for `Acions`
-  // which wouldn't find anythig else anyway.
+  // play if nothing else matches. E.g. a search for `AcIons`
+  // which wouldn't find anything else anyway.
```
Comment (jsoref) [inline src/search/middleware/es-search.js:189]: Misspelled words are problematic. I'm proposing a different corruption of `Actions` one which based on a fairly standard programming convention can be read as two words  (reading `AcIons` as `Ac` + `Ions`, where two letter strings are ignored as too short to be checked against dictionary words and `Ions` is a word in the dictionary) but which still won't find anything (and makes it easier for a spell checker to complain about finding `anythig`).

File: src/search/middleware/search.js [hunk 7]
```diff
@@ -79,7 +79,7 @@ async function handleGetSearchResultsError(req, res, error, options) {
     const reports = FailBot.report(error, Object.assign({ url: req.url }, options))
     // It might be `undefined` if no backends are configured which
     // is likely when using production NODE_ENV on your laptop
-    // where you might not have a HATSTACK_URL configured.
+    // where you might not have a HAYSTACK_URL configured.
```
Comment (jsoref) [inline src/search/middleware/search.js:82]: Sometimes misspelled words are intentional, or things reference obscure things, but here, I think it's just a typo

**Review by JoshSmith9419 — COMMENTED** [review]
I only have one *

**Comment by mchammer01** [comment]
@github/docs-engineering 👋🏻 - this fixes spelling mistakes mostly in scripts. Would you like to fix these internally as we appear to not be accepting external contributions to these files? 🤔

**Comment by mchammer01** [comment]
That said, the fixes look great to me, thanks @jsoref 🙇🏻 ✨

**Comment by jsoref** [comment]
Fwiw, there's a `sarch` in one of the `search` workflows -- https://github.com/github/docs/blob/4c0d5f11071f58853f682c2af0ee3e7cc0f11c54/.github/workflows/sync-search-pr.yml#L3

I called it out elsewhere and the public side seemed disinclined to deal with it.

(I might have other things that should be fixed in workflows as this PR was designed for just scripts which the PR system didn't explicitly say were forbidden and I'm trying to make these PRs reasonably small and scoped to limit how many reviews would be needed...)

**Review by heiskr — APPROVED** [review]
Thank you :) I think its okay to admin merge for these, its just typos, so I don't think there's any security concerns.

**Comment by peterbe** [comment]
> Thank you :) I think its okay to admin merge for these, its just typos, so I don't think there's any security concerns.

Makes me a bit nervous that `src/content-linter/tests/unit/liquid-data-tags.js` passed before despite the typo there. :)

**Comment by nguyenalex836** [comment]
> Thanks @jsoref for these! If you want to get a speedier review next time, consider breaking it up into multiple smaller PRs of ~10 files each.

That's on me! @jsoref and I discussed previously, and I didn't advise to break into smaller PRs. Thank you for the vast number of quality contributions, which I'm sure is no small amount of effort on your part 💛

**Comment by jsoref** [comment]
@peterbe: yeah, the liquid-data-tags test definitely worried me. Fwiw, it's a great example of why I encourage people to run a spell-checker on everything (including tests) -- I regularly run into tests that aren't testing what people think they're testing.

**Comment by jsoref** [comment]
(Oh, and props to the GitHub team. These PRs have been merged very quickly, even though they aren't tiny. I regularly encounter w/ projects that take over a year to merge smaller PRs.)

**Pull request merged**